[BUG] make lint error

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
Numerous errors occur when executing the `make lint ` command locally

```

$ make lint
/Users/bingyu/GolangProjects/bin/golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/bingyu/GolangProjects/src/github.com/openyurtio/openyurt /Users/bingyu/GolangProjects/src/github.com/openyurtio /Users/bingyu/GolangProjects/src/github.com /Users/bingyu/GolangProjects/src /Users/bingyu/GolangProjects /Users/bingyu /Users /]
INFO [config_reader] Used config file .golangci.yaml
INFO [lintersdb] Active 8 linters: [gci gofmt goimports govet ineffassign misspell staticcheck unused]
INFO [loader] Go packages loading at mode 575 (files|imports|name|types_sizes|compiled_files|exports_file|deps) took 3.405947066s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 35.975073ms
INFO [linters context/goanalysis] analyzers took 192.584189ms with top 10 stages: ctrlflow: 11.792221ms, SA5002: 9.654568ms, buildir: 8.633679ms, fact_deprecated: 5.630933ms, printf: 5.101242ms, goimports: 4.390829ms, misspell: 4.237856ms, gofmt: 3.790957ms, nilness: 3.344805ms, SA5012: 3.210551ms
INFO [runner/max_same_issues] 953/956 issues with text "undeclared name: `klog`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 451/454 issues with text "undeclared name: `Expect`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 130/133 issues with text "\"k8s.io/klog/v2\" imported but not used" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 129/132 issues with text "undeclared name: `It`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 44/47 issues with text "undeclared name: `ginkgo`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 28/31 issues with text "undeclared name: `Context`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 24/27 issues with text "undeclared name: `BeforeEach`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 23/26 issues with text "undeclared name: `By`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 14/17 issues with text "undeclared name: `AfterEach`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 9/12 issues with text "undeclared name: `Describe`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 8/11 issues with text "could not import k8s.io/apimachinery/pkg/types (-: could not load export data: cannot import \"k8s.io/apimachinery/pkg/types\" (unknown iexport format version 2), export data is newer version - update tool)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 4/7 issues with text "\"github.com/onsi/ginkgo/v2\" imported but not used" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 3/6 issues with text "undeclared name: `version`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 3/6 issues with text "undeclared name: `BeNil`" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "`v2` redeclared in this block" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "c.Lock undefined (type *componentKeyCache has no field or method Lock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "could not import k8s.io/apiserver/pkg/authentication/user (-: could not load export data: cannot import \"k8s.io/apiserver/pkg/authentication/user\" (unknown iexport format version 2), export data is newer version - update tool)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "c.Unlock undefined (type *componentKeyCache has no field or method Unlock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "\tother declaration of v2" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 2/5 issues with text "\"github.com/onsi/gomega\" imported but not used" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "coordinator.Unlock undefined (type *coordinator has no field or method Unlock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "coordinator.Lock undefined (type *coordinator has no field or method Lock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "c.Unlock undefined (type *CertManager has no field or method Unlock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "obj.Annotations undefined (type *NodePool has no field or method Annotations)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "f.Unlock undefined (type *FakePodControl has no field or method Unlock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "c.Lock undefined (type *CertManager has no field or method Lock)" were hidden, use --max-same-issues
INFO [runner/max_same_issues] 1/4 issues with text "f.Lock undefined (type *FakePodControl has no field or method Lock)" were hidden, use --max-same-issues
INFO [runner/max_from_linter] 158/208 issues from linter typecheck were hidden, use --max-issues-per-linter
INFO [runner] Issues before processing: 260590, after processing: 50
INFO [runner] Processors filtering stat (out/in): max_from_linter: 50/208, severity-rules: 50/50, cgo: 260590/260590, filename_unadjuster: 260590/260590, path_prettifier: 260590/260590, exclude: 260590/260590, max_same_issues: 208/2050, skip_dirs: 260590/260590, identifier_marker: 260590/260590, nolint: 260590/260590, diff: 2050/2050, path_shortener: 50/50, skip_files: 260590/260590, uniq_by_line: 2050/260590, max_per_file_from_linter: 2050/2050, path_prefixer: 50/50, sort_results: 50/50, autogenerated_exclude: 260590/260590, exclude-rules: 260590/260590, source_code: 50/50
INFO [runner] processing took 4.30076034s with stages: exclude-rules: 2.191052875s, identifier_marker: 1.724431278s, path_prettifier: 145.679235ms, nolint: 99.752122ms, filename_unadjuster: 45.039962ms, skip_dirs: 30.463934ms, cgo: 29.9822ms, uniq_by_line: 20.942234ms, autogenerated_exclude: 10.643661ms, source_code: 1.695684ms, max_same_issues: 717.791µs, max_per_file_from_linter: 291.872µs, max_from_linter: 43.874µs, path_shortener: 21.856µs, sort_results: 396ns, exclude: 322ns, diff: 301ns, severity-rules: 290ns, skip_files: 287ns, path_prefixer: 166ns
INFO [runner] linters took 17.928710009s with stages: goanalysis_metalinter: 13.62700642s
cmd/yurt-controller-manager/app/options/options.go:165:32: undeclared name: `klog` (typecheck)
	eventBroadcaster.StartLogging(klog.Infof)
	                              ^
cmd/yurt-controller-manager/app/options/options.go:40:2: "k8s.io/klog/v2" imported but not used (typecheck)
	"k8s.io/klog/v2"
	^
pkg/yurthub/kubernetes/serializer/serializer.go:200:2: undeclared name: `klog` (typecheck)
	klog.Infof("%#+v is not found in client-go runtime scheme", gvr)
	^
pkg/yurthub/kubernetes/serializer/serializer.go:226:5: sm.Lock undefined (type *SerializerManager has no field or method Lock) (typecheck)
	sm.Lock()
	   ^
pkg/yurthub/kubernetes/serializer/serializer.go:227:11: sm.Unlock undefined (type *SerializerManager has no field or method Unlock) (typecheck)
	defer sm.Unlock()
	         ^
pkg/yurthub/kubernetes/serializer/serializer.go:258:19: s.Decoder undefined (type *Serializer has no field or method Decoder) (typecheck)
	decoder, err = s.Decoder(mediaType, params)
	                 ^
pkg/yurthub/kubernetes/serializer/serializer.go:281:20: s.Encoder undefined (type *Serializer has no field or method Encoder) (typecheck)
	encoder, err := s.Encoder(mediaType, params)
	                  ^
pkg/yurthub/kubernetes/serializer/serializer.go:300:62: s.StreamDecoder undefined (type *Serializer has no field or method StreamDecoder) (typecheck)
	embeddedObjectDecoder, streamingSerializer, framer, err = s.StreamDecoder(mediaType, params)
	                                                            ^
pkg/yurthub/kubernetes/serializer/serializer.go:333:28: s.Encoder undefined (type *Serializer has no field or method Encoder) (typecheck)
	embeddedEncoder, err := s.Encoder(mediaType, params)
	                          ^
pkg/yurthub/kubernetes/serializer/serializer.go:366:2: undeclared name: `klog` (typecheck)
	klog.V(5).Infof("encode bytes data: write bytes=%d, size=%d, bytes=%v", n, sw.size, p[:n])
	^
pkg/yurthub/kubernetes/serializer/serializer.go:38:2: "k8s.io/klog/v2" imported but not used (typecheck)
	"k8s.io/klog/v2"
	^
pkg/controller/kubernetes/controller/controller_utils.go:28:2: could not import k8s.io/apimachinery/pkg/types (-: could not load export data: cannot import "k8s.io/apimachinery/pkg/types" (unknown iexport format version 2), export data is newer version - update tool) (typecheck)
	"k8s.io/apimachinery/pkg/types"
	^
pkg/controller/kubernetes/controller/util/node/controller_utils.go:27:2: could not import k8s.io/apimachinery/pkg/types (-: could not load export data: cannot import "k8s.io/apimachinery/pkg/types" (unknown iexport format version 2), export data is newer version - update tool) (typecheck)
	"k8s.io/apimachinery/pkg/types"
	^
pkg/controller/kubernetes/controller/util/node/controller_utils.go:32:2: "k8s.io/klog/v2" imported but not used (typecheck)
	"k8s.io/klog/v2"
	^
pkg/controller/util/node/controller_utils.go:28:2: could not import k8s.io/apimachinery/pkg/types (-: could not load export data: cannot import "k8s.io/apimachinery/pkg/types" (unknown iexport format version 2), export data is newer version - update tool) (typecheck)
	"k8s.io/apimachinery/pkg/types"

```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- OpenYurt version:
- Kubernetes version (use `kubectl version`):
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**others**

/kind bug

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -43,7 +43,7 @@ jobs:
       - name: Lint golang code
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.51.2
+          version: v1.47.3
           args: -v
 
   markdownlint-misspell-shellcheck:

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,10 @@
 # limitations under the License.
 
 KUBERNETESVERSION ?=v1.22
+GOLANGCILINT_VERSION ?= v1.47.3
+GLOBAL_GOLANGCILINT := $(shell which golangci-lint)
+GOBIN := $(shell go env GOPATH)/bin
+GOBIN_GOLANGCILINT := $(shell which $(GOBIN)/golangci-lint)
 TARGET_PLATFORMS ?= linux/amd64
 IMAGE_REPO ?= openyurt
 IMAGE_TAG ?= $(shell git describe --abbrev=0 --tags)
@@ -91,14 +95,18 @@ e2e-tests:
 	ENABLE_AUTONOMY_TESTS=${ENABLE_AUTONOMY_TESTS} TARGET_PLATFORMS=${TARGET_PLATFORMS} hack/make-rules/run-e2e-tests.sh
 
 install-golint: ## check golint if not exist install golint tools
-ifeq (, $(shell which golangci-lint))
-	@{ \
-	set -e ;\
-	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.51.2 ;\
-	}
-GOLINT_BIN=$(shell go env GOPATH)/bin/golangci-lint
+ifeq ($(shell $(GLOBAL_GOLANGCILINT) version --format short), $(GOLANGCILINT_VERSION))
+GOLINT_BIN=$(GLOBAL_GOLANGCILINT)
+else ifeq ($(shell $(GOBIN_GOLANGCILINT) version --format short), $(GOLANGCILINT_VERSION))
+GOLINT_BIN=$(GOBIN_GOLANGCILINT)
 else
-GOLINT_BIN=$(shell which golangci-lint)
+	@{ \
+    set -e ;\
+    echo 'installing golangci-lint-$(GOLANGCILINT_VERSION)' ;\
+    go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(GOLANGCILINT_VERSION) ;\
+    echo 'Successfully installed' ;\
+    }
+GOLINT_BIN=$(GOBIN)/golangci-lint
 endif
 
 lint: install-golint ## Run go lint against code.

